Manage Policies Page: New policy modal

RachelElysia opened the pull request:
Cerra #2594 

- New policy modal gives users the options of choosing from 8 prefixed policies or creating one of their own
- New policy modal is no longer based on queries
- Query a host on host details page revamped to match new frontend patterns (remove redux, remove PropTypes, etc)

https://www.loom.com/share/3bcbd46a04a5496db2d94f8f179a155c

Screenshot:
<img width="1299" alt="Screen Shot 2021-11-24 at 12 54 34 PM" src="https://user-images.githubusercontent.com/71795832/143290506-99c09782-a107-470f-9c00-dccfbcd819de.png">

# Checklist for submitter

If some of the following don't apply, delete the relevant line.

- [x] Changes file added (for user-visible changes)
~~- [ ] Documented any API changes (docs/01-Using-Fleet/03-REST-API.md)~~
~~- [ ] Documented any permissions changes~~
~~- [ ] Added/updated tests~~ On different issue
- [x] Manual QA for all new/changed functionality

noahtalerman commented:
@RachelElysia below are tweaks to several of the policies' queries so that they correctly return "Yes" and "No" answers. Can you please add these changes in your PR here? Thanks :)

#### Is System Integrity Protection (SIP) enabled for macOS devices
Update this policy's `query` to the following:
`SELECT 1 FROM sip_config WHERE config_flag = 'sip' AND enabled=1`

`enabled=0` is now `enabled=1` so that hosts who answer "Yes" to the policy have the SIP feature enabled.

#### Are guest users not activated macOS devices?
Update this policy's `query` to the following:
`SELECT 1 FROM managed_policies WHERE domain = 'com.apple.loginwindow' AND name = 'com.apple.login.mcx.DisableAutoLoginClient' AND value=1 LIMIT 1`

`enabled=0` is now `enabled=1` so that hosts that answer "Yes" to the policy have the guest users disabled.

#### Is secure keyboard entry enabled for macOS devices?
Update this policy's `query` to the following:
`SELECT 1 FROM managed_policies WHERE domain = 'com.apple.Terminal AND name = 'SecureKeyboardEntry' AND value=1 LIMIT 1`

`value=0` is now `value=1` so that hosts that answer "Yes" to the policy have secure keyboard entry enabled.

lukeheath commented on frontend/pages/queries/QueryPage/QueryPage.tsx:
```diff
@@ -41,6 +41,10 @@ const QueryPage = ({
 }: IQueryPageProps): JSX.Element => {
   const queryIdForEdit = paramsQueryId ? parseInt(paramsQueryId, 10) : null;
+
+  console.log("paramsQueryId", paramsQueryId);
+  console.log("queryIdForEdit QueryPage", queryIdForEdit);
```
🪵

RachelElysia replied:
lol 💩

lukeheath reviewed (changes requested):
Looks great! Some small things: 

1) In the "Add a policy" modal, the last two template options will open the template, but the "Save" button does not All the other templates work as expected. 
- Side note, should it be "Are guests not activated _for/on_ macOS devices"?
- Side side note, we're inconsistent in our usage of for/on in the template names.

2) In the save policy modal, what do you think of setting the textarea styling to `line-height: 1.4;`? Right now it looks like too much vertical space: 

![image](https://user-images.githubusercontent.com/2495927/144499797-0befa309-a303-4158-ad43-ebf9896c5521.png)

3) When I have "All Teams" selected in the dropdown, it only shows me global policies. I have to select the team from the dropdown to see team policies. 

4) When I create my own policy, the resolution I enter into the save policy modal is not applied.

RachelElysia commented on frontend/pages/policies/ManagePoliciesPage/ManagePoliciesPage.tsx:
```diff
@@ -1,3 +1,5 @@
+// TODO: Rebase once Martavis is merged, delete add policy modal, replace with renamed AddPolicyModal
+
```
Whoops!

RachelElysia commented:
> 1. In the "Add a policy" modal, the last two template options will open the template, but the "Save" button does not open the save modal. All the other templates work as expected.
I fixed it, sorry, copied not working SQL from figma, fixed SQL in code and on figma

> * Side note, should it be "Are guests not activated _for/on_ macOS devices"?
> * Side side note, we're inconsistent in our usage of for/on in the template names.
I switched it all to "on", updated figma accordingly, and left a comment on figma that I switched the inconsistent verbiage

> 2. In the save policy modal, what do you think of setting the textarea styling to `line-height: 1.4;`? Right now it looks like too much vertical space:
Great call! I switched it to 1.5. 1.4 looked smushy to me, but yeah this should definitely be relative and not static pixels incase we use input fields with different font sizes elsewhere.

> ![image](https://user-images.githubusercontent.com/2495927/144499797-0befa309-a303-4158-ad43-ebf9896c5521.png)
> 
> 3. When I have "All Teams" selected in the dropdown, it only shows me global policies. I have to select the team from the dropdown to see team policies.
That's the correct UI no?

> 4. When I create my own policy, the resolution I enter into the save policy modal is not applied.
Weird, just tried this and mine applied-- Can you QA this again?

lukeheath reviewed (approved):
✨ 

>That's the correct UI no?
- You're right, this is the correct functionality. 

>Weird, just tried this and mine applied-- Can you QA this again?
- It's working for me now!
